lib/tasks/myturn_import.rake

```diff
@@ -0,0 +1,136 @@
+# For reference, these are the columns in a complete export (shown in alphabetical order):
+#
+# "Additional Image"
+# "Admin Notes"
+# "Agreement that must be signed to checkout the item"
+# "Amount / Fee"
+# "Attachment"
+# "Author"
+# "Between Buffer Days"
+# "Categories"
+# "Color"
+# "Condition"
+# "Current Location"
+# "Daily Late Fee"
+# "Date Created"
+# "Date Last Edited"
+# "Date Last Updated"
+# "Date Purchased"
+# "Default Loan Length"
+# "Default Scheduled Maintenance Plan"
+# "Description"
+# "Dimensions (WxHxD)"
+# "Eco Rating"
+# "Embodied Carbon"
+# "Emission Factor"
+# "Excluded Fulfillment Methods"
+# "Featured"
+# "Floating"
+# "Goes Home"
+# "Grace period on late fees (in days)"
+# "Historical Cost"
+# "Home Location"
+# "Image"
+# "Item ID"
+# "Item Type"
+# "Keywords"
+# "Location Code"
+# "Maintenance Plan After Each Check In"
+# "Manufacturer"
+# "Max Frequency Units"
+# "Max Frequency"
+# "Max Output"
+# "Maximum number of renewals"
+# "Maximum number that may be reserved"
+# "Maximum percentage of inventory that may be reserved"
+# "Maximum Reservation Length"
+# "Min Frequency Units"
+# "Min Frequency"
+# "Model"
+# "Name"
+# "Now Buffer Days"
+# "Post-Description Text"
+# "Pre-Description Text"
+# "Product Code / UPC"
+# "Publisher"
+# "Purchased"
+# "Replacement Cost"
+# "Serial Number"
+# "Size"
+# "Source / Supplier"
+# "Status(es)"
+# "Tax(es)"
+# "Weight"
+
+namespace :myturn do
+  task reset: :environment do
+    # Just for development
+    ReservationLoan.delete_all
+    ReservationHold.delete_all
+    ReservableItem.delete_all
+    ItemPool.delete_all
+  end
+
+  task import: :environment do
+    path = ENV["CSV_FILE"]
+    library = Library.find(ENV["LIBRARY_ID"])
+    user = User.find_by!(email: ENV["AUTHOR"])
+
+    successful_imports = 0
+    failed_imports = 0
+    failures = []
+    non_numeric_ids = 0
+
+    CSV.foreach(path, headers: true) do |row|
+      Audited.audit_class.as_user(user) do
+        ActsAsTenant.with_tenant(library) do
+          item_number = begin
+            Integer(row["Item ID"])
+          rescue
+            non_numeric_ids += 1
+            next
+          end
+
+          item_pool = ItemPool.create_with(creator: user).find_or_create_by!(name: row["Name"])
+
+          item_pool.update(
```

Comment:
Might be helpful to make this `item.update!`. Right now this update can fail silently. (although this one isn't caught by the `rescue` below)

Reply:
I made this `update!` and moved it and the statement above into the `begin`/`rescue` block so that the error handling is more consistent.